lib/ecl/ecl_grid.cpp

```diff
@@ -3000,10 +3002,25 @@ static ecl_grid_type * ecl_grid_alloc_EGRID__( ecl_grid_type * main_grid , const
     eclipse_version = main_grid->eclipse_version;
   }
 
-
-  /** If ACTNUM is not present - that is is interpreted as - all active. */
-  if (ecl_file_get_num_named_kw(ecl_file , ACTNUM_KW) > grid_nr)
+  // If ACTNUM and ext_actnum are not present - that is is interpreted as - all active. 
+  const int * actnum_data = NULL;
+  if (ecl_file_get_num_named_kw(ecl_file , ACTNUM_KW) > grid_nr) {
     actnum_kw = ecl_file_iget_named_kw( ecl_file , ACTNUM_KW    , grid_nr);
+    actnum_data = ecl_kw_get_int_ptr(actnum_kw);
+  }
```

Comment:
I would continue the product block inside the `if (ecl_file_get_num_named_kw(ecl_file, ACTNUM_KW) > grid_nr)` block.